### PR TITLE
[CALCITE-6549] Add LOG1P function (enabled in Spark library)

### PR DESCRIPTION
**Pull request by caicancai**
https://issues.apache.org/jira/browse/CALCITE-6549

### CODE REVIEW

**Comment by caicancai** [comment]
@NobiGo PTAL, thanks

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 1]
```diff
@@ -4292,18 +4294,27 @@ private static List<Expression> args(RexCall call,
       operand0 = operands.left;
       operand1 = operands.right;
       boolean nonPositiveIsNull = library == SqlLibrary.MYSQL ? true : false;
```
Comment (NobiGo) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:4296]: We should pass the library as the Parameter. Then the code logic will be clear. WDYT?

Comment (NobiGo) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:4296]: Maybe we should refactor the code in another issue. The current code here has poor extensibilit.

Comment (caicancai) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:4296]: Yes, I have to admit this. https://issues.apache.org/jira/browse/CALCITE-6550 is already working on this. I'll close it first.

Comment (caicancai) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:4296]: @NobiGo Could you please post your opinion at https://issues.apache.org/jira/browse/CALCITE-6550? I think it is a very complicated issue.

Comment (julianhyde) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:4296]: I don't think you should pass in a library at run time. The runtime should not be worrying about high-level concepts like llibrary.

**Pull request closed without merging**